### PR TITLE
#5072 Marketplace sidebar activation UI fixes

### PR DESCRIPTION
**Pull request by BLoe**
## What does this PR do?

- Follow up from #5072 - UI polish

## Demo

- _Paste a screenshot or demo video here_

## Checklist

- [ ] Add tests
- [ ] Designate a primary reviewer

### CODE REVIEW

File: src/utils/strings.ts [hunk 1]
```diff
@@ -16,9 +16,10 @@
  */
 
 /**
- * Link to the marketplace with trailing slash
+ * Link to the marketplace
  */
-export const MARKETPLACE_URL = "https://www.pixiebrix.com/marketplace/";
+// eslint-disable-next-line prefer-destructuring -- breaks EnvironmentPlugin
+export const MARKETPLACE_URL = process.env.MARKETPLACE_URL;
```
Comment (BLoe) [inline src/utils/strings.ts:22]: This isn't working right now, getting "Cannot read properties of undefined (reading 'MARKETPLACE_URL')"

File: webpack.config.js [hunk 2]
```diff
@@ -105,7 +107,7 @@ const sourceMapPublicUrl =
   `${process.env.SOURCE_MAP_URL_BASE}/${process.env.SOURCE_MAP_PATH}/`;
 console.log(
   "Sourcemaps:",
-  sourceMapPublicUrl ? sourceMapPublicUrl : produceSourcemap ? "Local" : "No"
+  sourceMapPublicUrl ?? produceSourcemap ? "Local" : "No"
```
Comment (twschiller) [inline webpack.config.js:110]: Doesn't this change the behavior due to ?? having higher precedence? In any case, it's very hard to read

File: src/components/ServiceAuthSelector.tsx [hunk 3]
```diff
@@ -92,11 +92,6 @@ const ServiceAuthSelector: React.FunctionComponent<{
     return (
       <Form.Group controlId={field.name}>
         <Form.Control type="text" readOnly value="Automatic" />
-        {meta.error && (
```
Comment (twschiller) [inline src/components/ServiceAuthSelector.tsx:95]: What is this change for?

File: webpack.config.js (hunk 2 above)
Comment (BLoe) [reply webpack.config.js:110]: I was getting a lint warning about the ternary. Should I just revert this change?

File: src/options/pages/marketplace/ActivateWizard.tsx [hunk 4]
```diff
@@ -109,14 +109,14 @@ const ActivateWizard: React.FunctionComponent<OwnProps> = ({ blueprint }) => {
   const action = reinstall ? "Reactivate" : "Activate";
   useTitle(`${action} ${truncate(blueprint.metadata.name, { length: 15 })}`);
 
-  const authPersistenceContext: ReduxPersistenceContextType = {
+  const reduxPersistenceContext: ReduxPersistenceContextType = {
```
Comment (twschiller) [inline src/options/pages/marketplace/ActivateWizard.tsx:112]: The naming here may want to reference that it's the persistence for what looks like the optionsSlice: https://github.com/pixiebrix/pixiebrix-extension/pull/5200/files#diff-f2f244dbc0fd870e51a233cb1bdb883283b329907b61be4fd4d277a82e42eccdR40

File: src/components/ServiceAuthSelector.tsx (hunk 3 above)
Comment (BLoe) [reply src/components/ServiceAuthSelector.tsx:95]: `ServiceBody` and `ServiceRow` have error UI handling now, we don't need to show it here.

**Review by twschiller — APPROVED** [review]

File: src/options/pages/marketplace/ActivateWizard.tsx (hunk 4 above)
Comment (BLoe) [reply src/options/pages/marketplace/ActivateWizard.tsx:112]: It's the options store not the options slice. The persistor is at the store level.

File: src/utils/strings.ts (hunk 1 above)
Comment (BLoe) [reply src/utils/strings.ts:22]: never mind, needed to restart webpack 🤦

**Comment by BLoe** [comment]
@twschiller @BALEHOK I'm going to merge this in order to get a build up. Feel free to leave more comments/review on this and I can address in a follow-up PR.

**Pull request merged**